src/server/services/post.service.ts

```diff
@@ -114,53 +158,65 @@ export const getPostsInfinite = async ({
       return { items: [] };
     }
 
-    const collectionItemModelsAND: Prisma.Enumerable<Prisma.CollectionItemWhereInput> =
-      getAvailableCollectionItemsFilterForUser({ permissions, user });
```

Comment:
Also, you still need this, just in plain SQL like in the images. This is what helps filter out collection items that are accepted/review/rejected for the public view

The Query looks like this:
```sql
const displayReviewItems = userId
      ? `OR (ci."status" = 'REVIEW' AND ci."addedById" = ${userId})`
      : '';

    AND.push(Prisma.sql`EXISTS (
      SELECT 1 FROM "CollectionItem" ci
      WHERE ci."collectionId" = ${collectionId}
        AND ci."postId" = p.id
        AND (ci."status" = 'ACCEPTED' ${Prisma.raw(displayReviewItems)})
    )`);
```